Extensions/CursorType.json

```diff
@@ -36,26 +36,125 @@
           "name": "style_cursor",
           "optional": false,
-          "supplementaryInformation": "",
-          "type": "string"
+          "supplementaryInformation": "[\"auto\", \"default\", \"none\", \"context-menu\", \"help\", \"pointer\", \"progress\", \"wait\", \"cell\", \"crosshair\", \"text\", \"vertical-text\", \"alias\", \"copy\", \"move\", \"no-drop\", \"not-allowed\", \"grab\", \"grabbing\", \"all-scroll\", \"col-resize\", \"row-resize\", \"n-resize\", \"e-resize\", \"s-resize\", \"w-resize\", \"ne-resize\", \"nw-resize\", \"se-resize\", \"sw-resize\", \"ew-resize\", \"ns-resize\", \"nesw-resize\", \"nwse-resize\", \"zoom-in\", \"zoom-out\"]",
+          "type": "stringWithSelector"
         }
       ],
       "objectGroups": []
     }
   ],
-  "eventsBasedBehaviors": []
+  "eventsBasedBehaviors": [
+    {
+      "description": "Sets the cursor appearence on hover of the object.",
+      "fullName": "Hover cursor",
+      "name": "CursorHover",
+      "objectType": "",
+      "eventsFunctions": [
+        {
+          "description": "",
+          "fullName": "",
+          "functionType": "Action",
+          "name": "doStepPreEvents",
+          "sentence": "",
+          "events": [
+            {
+              "disabled": false,
+              "folded": false,
+              "type": "BuiltinCommonInstructions::JsCode",
+              "inlineCode": "objects[0]\n    .getRendererObject()\n    .cursor = objects[0]\n        .getBehavior(\n            eventsFunctionContext.getBehaviorName(\"Behavior\")\n        )\n        ._getcursor_type();\n",
+              "parameterObjects": "Object",
+              "useStrict": true
+            }
+          ],
+          "parameters": [
+            {
+              "codeOnly": false,
+              "defaultValue": "",
+              "description": "Object",
+              "longDescription": "",
+              "name": "Object",
+              "optional": false,
+              "supplementaryInformation": "",
+              "type": "object"
+            },
+            {
+              "codeOnly": false,
+              "defaultValue": "",
+              "description": "Behavior",
+              "longDescription": "",
+              "name": "Behavior",
+              "optional": false,
+              "supplementaryInformation": "CursorType::CursorHover",
+              "type": "behavior"
+            }
+          ],
+          "objectGroups": []
+        },
+        {
+          "description": "",
+          "fullName": "",
+          "functionType": "Action",
+          "name": "onCreated",
+          "sentence": "",
+          "events": [
+            {
+              "disabled": false,
+              "folded": false,
+              "type": "BuiltinCommonInstructions::JsCode",
+              "inlineCode": "const renderer = objects[0].getRendererObject();\nrenderer.interactive = true; // Activate necessary pixi plugin\n",
```

Comment:
Usually in the game engine we're never using PIXI `interactive` because I think it runs a separate timer and logic to check for hovering/collisions. More generally, we use PIXI only as a rendering engine. The exception is for the IDE where we use the interaction capabilities.

Could you see if you can rework the behavior to use `isCollidingWithPoint` or the equivalent condition checking if a point is inside an object?
You'll need to get the cursor position with MouseX, MouseY - don't forget to pass the layer of the object.

I would like to centralize everything that is related to collisions through the game engine - because later this can be reworked to use optimizations like spatial partitioning.

Reply:
> don't forget to pass the layer of the object.

I don't see where I should pass the object layer? Neither MouseX, MouseY or isCollidingWithPoint seem to require it. It seems to work anyways.

Reply:
The `gdjs.InputManager` getMouseX/getMouseY returns the position in the coordinates of the canvas.
`isCollidingWithPoint` is working with what we usually call "world coordinates", i.e: the real position of the objects.

The canvas coordinates and the world coordinates are only the same if your camera is not moved. As soon as your camera is moved/rotated/zoomed, they are not the same at all anymore :) 
Fortunately, we offer a way to translate from one to the other: [`convertCoords` on `gdjs.Layer`](https://docs.gdevelop-app.com/GDJS%20Runtime%20Documentation/gdjs.Layer.html#convertCoords)

Get the layer name of the object, get the layer from the runtimescene, call this with the mouse x/y position (and camera id 0) and send the returned coordinates to isCollidingWithPoint and you should be good. You can check that by moving the camera in your game and see that the object with the behavior still has a cursor :)